zerver/views/auth.py

```diff
@@ -468,3 +468,15 @@ def api_fetch_google_client_id(request):
 def logout_then_login(request, **kwargs):
     # type: (HttpRequest, **Any) -> HttpResponse
     return django_logout_then_login(request, kwargs)
+
+
+def get_websocket_auth_data(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
+    login(request, user_profile, 'zproject.backends.ZulipDummyBackend')
+    return json_success({
+        'websocket_auth_credentials':
+            {
+                settings.CSRF_COOKIE_NAME: get_token(request),
+                settings.SESSION_COOKIE_NAME: request.session.session_key
+            }
+        })
```

Comment:
So, what I had in mind was actually modifying this section of zerver/tornado/socket.py: 
```
        user_profile = get_user_profile(self.browser_session_id)                                                 
        if user_profile is None:                                                                                 
            raise SocketAuthError('Unknown or missing session')                                                  
        self.session.user_profile = user_profile                                                                 
                                                                                                                 
        if not _compare_salted_tokens(msg['request']['csrf_token'], self.csrf_token):                            
            raise SocketAuthError('CSRF token does not match that in cookie')                                    
```

to support getting the user_profile object to set self.session.user_profile via the API key lookup mechanism (see examples in zerver/decorator.py), rather than a CSRF token + session cookie.  Does that make sense?